cfgov/unprocessed/apps/paying-for-college/js/disclosures/dispatchers/get-api-values.js

```diff
@@ -130,16 +89,14 @@ const getApiValues = {
     const warning = document
       .querySelector('[data-warning]')
       .getAttribute('data-warning');
-    if (warning !== '' && typeof warning !== 'undefined') {
+    if (warning && !window.Cypress) {
```

Comment:
Should this code know about Cypress? It seems like whatever the need is here should be offloaded to the test logic.

Reply:
Yeah this is kinda gross, the problem is though that the data isn't available in the test db, so this warning code will always trigger, even though the actual http requests are stubbed.

Putting this logic here allows Cypress to skip the warning code (which we want) and, in normal use of the app, allows the warning conditional to trigger when we need it.

I think maybe the best solution would be rearchitecting how the warning logic works/how the warning is propagated from the backend to the frontend, but that seemed out of scope.

Reply:
Yeah, I'd be happy to fast-follow that because this is pretty gross. 😁